distribution/lib/Standard/Base/0.2.32-SNAPSHOT/src/Data/Vector.enso

```diff
@@ -127,14 +127,35 @@ type Vector
     ## Gets an element from the vector at a specified index (0-based).
 
        Arguments:
-       - index: The location in the vector to get the element from.
+       - index: The location in the vector to get the element from. The index is
+         also allowed be negative, then the elements are indexed from the back
+         of the vector, i.e. -1 will correspond to the last element.
 
        > Example
          Get the second element of a vector.
 
-             [1, 2, 3].at 1
-    at : Number -> Any
-    at index = this.to_array.at index
+             [1, 2, 3].at 1 == 2
+
+       > Example
+         Get the last element of a vector.
+
+             [1, 2, 3].at -1 == 3
+    at : Integer -> Any ! Index_Out_Of_Bounds_Error
+    at index =
+        actual_index = if index < 0 then this.length + index else index
+        if actual_index>=0 && actual_index<this.length then this.to_array.at actual_index else
```

Comment:
why bounds check here if you already have a check in the underlying array?

Reply:
As we discussed, we need to either bounds check or do a `Panic.recover` to catch the underlying array error and rethrow it as an Enso dataflow error.
Given that we agreed that both seem to have the same cost, I kept the current implementation (this was done before I fixed the `ArrayGetAtNode`).
Do you think we should switch to `Panic.recover`?

```suggestion
        Panic.recover (this.to_array.at actual_index) . catch err-> case err of
            Invalid_Array_Index_Error ->
```

Reply:
Changed to a `catch`, but it is a bit of a hassle to correctly differentiate panics from the array access from dataflow errors which may be stored within index.

```
    at : Integer -> Any ! Index_Out_Of_Bounds_Error
    at index =
        actual_index = if index < 0 then this.length + index else index
        case actual_index of
            _ ->
                Panic.recover (this.to_array.at actual_index) . catch err-> case err of
                    Invalid_Array_Index_Error _ _ -> Error.throw (Index_Out_Of_Bounds_Error index this.length)
                    _ -> Panic.throw err
```

Reply:
I needed to revert this change, because I needed to handle the following test cases (breaking them also broke `Visualization_Tests`):
```
    Test.specify "should allow to store dataflow errors and raise them on access" <|
        vec = [Error.throw (My_Error "foo"), "bar"]
        vec.at 1 . should_equal "bar"
        vec.at 0 . should_fail_with My_Error
```

The visualization tests rely on being able to store dataflow errors inside of the vector, but raising them upon access.

The solution with Panic.recover is unable to distinguish a different Panic raised from within the access operation from dataflow errors raised because we access a stored error, so it does not know at which level to rethrow. I summarised the issue on Discord. This example shows that we **really** need a specialized `catch` variant which works only for Panics and allows us to distinguish and handle Panics separately from dataflow errors.

Reply:
I suggest we keep the check for now and revisit this once we decide on how we should handle and distinguish Panics from Dataflow Errors. 

I will add a TODO in the code.